src/pybind/mgr/dashboard/frontend/src/app/shared/services/notification.service.ts

```diff
@@ -63,17 +64,43 @@ export class NotificationService {
     localStorage.setItem(this.KEY, JSON.stringify(recent));
   }
 
+  queueNotifications(notifications: CdNotificationConfig[], ms: number = 500) {
+    this.queuedNotifications = this.queuedNotifications.concat(notifications);
+    window.setTimeout(() => {
+      this.sendQueuedNotifications();
+    }, ms);
+  }
+
+  private sendQueuedNotifications() {
+    _.uniqWith(this.queuedNotifications, _.isEqual).forEach((notification) => {
+      this.show(notification);
+    });
+    this.queuedNotifications = [];
+  }
+
   /**
    * Method for showing a notification.
    * @param {NotificationType} type toastr type
    * @param {string} title
    * @param {string} [message] The message to be displayed. Note, use this field
    *   for error notifications only.
    * @param {*} [options] toastr compatible options, used when creating a toastr
-   * @memberof NotificationService
    * @returns The timeout ID that is set to be able to cancel the notification.
    */
-  show(type: NotificationType, title: string, message?: string, options?: any) {
+  show(type: NotificationType, title: string, message?: string, options?: any): number;
+  show(config: CdNotificationConfig): number;
+  show(
+    arg: NotificationType | CdNotificationConfig,
+    title?: string,
+    message?: string,
+    options?: any
+  ): number {
+    let type;
+    if (_.isObject(arg)) {
+      ({ message, type, title, options } = <CdNotificationConfig>arg);
+    } else {
+      type = arg;
+    }
     return setTimeout(() => {
```

Comment:
Could you apply the following change?
Since you are now returning a number and tslint thinks setTimeout is a node function it is showing an error in my IDE.

```suggestion
     return window.setTimeout(() => {
```